project/Dependencies.scala

```diff
@@ -109,7 +109,11 @@ object ExternalDependencies {
     val logstashLogback = "6.1"
     val scribeJava = "6.8.1"
     val apm = "1.11.0"
+    val enumeratumVersion = "1.5.13"
   }
+  val enumeratumDependencies = Seq(
+    "com.beachape" %% "enumeratum" % versions.enumeratumVersion
```

Comment:
Looking at the docs for this [here](https://github.com/lloydmeta/enumeratum).. It's not clear to me what this gives on top of using plain old sealed traits and case objects

e.g.

```scala
sealed trait WorkType

object WorkType {
  case object Audio extends WorkType
  case object EBook extends WorkType
  ...
}
```

All of the things it mentions (dependencies, all compile time, performance etc) are just as good (or even better) without this, is there sometime I am missing?

Reply:
It precomputes the possible `values`. So that there doesn't need to be an enumeration of all the objects somewhere to be able to write the `fromCode` function (or at least I don't know how to do that with objects and sealed traits)

Reply:
:+1: that makes sense

Reply:
This is a great bit of functionality for JSON deserialising!